ide/db/src/org/netbeans/api/db/explorer/ConnectionManager.java

```diff
@@ -187,6 +189,19 @@ public void removeConnection(DatabaseConnection dbconn) throws DatabaseException
 
         ConnectionList.getDefault().remove(dbconn.getDelegate());
     }
+    
+    /**
+     * Retrieves a list of database schemas available for given connection.
+     * 
+     * @param dbconn
+     * @return List of schemas or null if connection failed
+     */
+    public List<String> getSchemas(DatabaseConnection dbconn) {
+        if (SwingUtilities.isEventDispatchThread()) {
+            throw new IllegalStateException("This method can not be called on the event dispatch thread."); // NOI18N
```

Comment:
_cannot_ is a single word - maybe you want to spell this (unusual?) restriction in the Javadoc of the method.